src/Base/Vector3D.cpp

```diff
@@ -437,6 +437,17 @@ Vector3<float_type>& Vector3<float_type>::Normalize()
     return *this;
 }
 
+template<class float_type>
+Vector3<float_type> Vector3<float_type>::getNormalized() const
+{
+    float_type fLen = Length();
+    if (fLen != static_cast<float_type>(0.0) && fLen != static_cast<float_type>(1.0)) {
+        return Vector3(x / fLen, y / fLen, z / fLen);
+    }
+
+    return Vector3(x, y, z);

```

Comment:
Why not simply

```
Vector3 copy = *this;
copy.Normalize();

return copy;
```

?

Reply:
I don't think so. They should be freecad when the parent node is deleted.

Reply:
Though, if I forgot to add something somewhere to the tree after allocating then that is a leak.